src/MessagePack/Resolvers/DynamicObjectResolver.cs

```diff
@@ -313,9 +313,9 @@ namespace MessagePack.Internal
     internal static class DynamicObjectTypeBuilder
     {
 #if NETFRAMEWORK || NETSTANDARD2_0
-        private static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", RegexOptions.Compiled);
+        internal static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", RegexOptions.Compiled);
 #else
-        private static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported ? RegexOptions.Compiled : RegexOptions.None);
+        internal static readonly Regex SubtractFullNameRegex = new(@", Version=\d+.\d+.\d+.\d+, Culture=\w+, PublicKeyToken=\w+", System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported ? RegexOptions.Compiled : RegexOptions.None);
```

Comment:
Better. But what I actually meant was to reuse the `AvoidDynamicCode` field in the MessagePackSerializer.NonGeneric.cs file by making _that_ `internal`. That way even this particular site won't require `#if` on it.

Reply:
DynamicCodeDumper csproj does not include MessagePackSerializer.*cs.
Should I make a new internal utility class?

Reply:
Maybe that would be ideal, yes. Then have both consumers use that utility class.